src/gui/sdlmain.cpp

```diff
@@ -773,17 +786,106 @@ static bool LoadGLShaders(const char *src, GLuint *vertex, GLuint *fragment) {
 }
 #endif
 
-Bitu GFX_SetSize(Bitu width,Bitu height,Bitu flags,double scalex,double scaley,GFX_CallBack_t callback) {
+/* Get the maximum area available for drawing: */
+/* TODO: The current implementation will not let the pixel-perfect mode to  */
+/*       to apply even the simplest 1:2 and 2:1 corrections at the original */
+/*       window resolution, therefore: consider a special case for PP mode. */
+static void GetAvailableArea(int *width, int *height)
+{
+	bool   fixed;
+	double par;
+
+	fixed = false;
+	if (sdl.desktop.fullscreen) {
+		if (sdl.desktop.full.fixed) {
+			*width  = sdl.desktop.full.width;
+			*height = sdl.desktop.full.height;
+			fixed   = true;
+		}
+	}
+	else {
+		if (sdl.desktop.window.width > 0) {
+			*width  = sdl.desktop.window.width;
+			*height = sdl.desktop.window.height;
+			fixed   = true;
+		}
+	}
+	if (!fixed) {
+		par = sdl.draw.par;
+		if (par > 1.0)
+			*height = round( *height * par );
+		if (par < 1.0)
+			*width  = round( *width  / par ); 
+	}
+}
+
+// ATT: aspect is the final aspect ratio of the image including its pixel dimensions and PAR
+/*static void GetActualArea( Bit16u av_w, Bit16u av_h, Bit16u *w, Bit16u *h, double aspect )
+{	double as_x, as_y;
+
+	if( aspect > 1.0 )
+	{	as_y =     aspect; as_x = 1.0;  } else
+	{	as_x = 1.0/aspect; as_y = 1.0;  }
+
+	if( av_h / as_y < av_w / as_x )
+	{	*h = av_h; *w = round( (double)av_h / aspect );  } else
+	{	*w = av_w; *h = round( (double)av_w * aspect );  }
+}*/
```

Comment:
This is a stub for better handling non-fixed resolution. For example, if the original resolution is 320x400, it will be automatically corrected to a minimum of 640x400. May I keep it lest I forget to implement that feature? If not, where can I record a *TODO*?

Reply:
OK.